hack/whitespace.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+#
+# Check for trailing whitespaces in all tracked files.
+#
+# Usage:
+# hack/whitespace.sh check # Check for trailing whitespaces
+# hack/whitespace.sh format # Drop trailing whitespaces
+
+set -e
+
+function format() {
+    git ls-files -- ':!vendor/' | xargs sed --follow-symlinks -i 's/[[:space:]]*$//'
+}
+
+function check() {
+    invalid_files=$(git ls-files -- ':!vendor/' | xargs egrep -Hn " +$" || true)
+    if [[ $invalid_files ]]; then
+        echo 'Found trailing whitespaces. Please remove trailing whitespaces using `make format`:'
+        echo "$invalid_files"
+        return 1
+    fi
+}
+
+if [ "$1" == "format" ]; then
+    format
+elif [ "$1" == "check" ]; then
+    check
+else
```

Comment:
if [[ "$1" =~ (check|format) ]]; then
   $1
else

Reply:
That saves line but pushes API from one layer (user params) to another (internal naming of functions). I'd prefer keeping the mapping as a dividing line between the two.

Reply:
ack